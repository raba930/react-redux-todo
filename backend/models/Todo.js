import mongoose from 'mongoose';
const Schema = mongoose.Schema;

const Todo = new Schema({
    token: String,
    id: Number,
    text: String,
    info: String,
    completed: Boolean,
    created: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Todo', Todo);
