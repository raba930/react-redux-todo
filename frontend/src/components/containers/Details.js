import { connect } from 'react-redux';
import * as actions from '../../actions/account';
import Details from '../views/Details';

const mapStateToProps = (state, prop) => {
    const todo = state.account.todos.find(todo => todo._id === prop.match.params.id);
    return {
        todo: todo
    };
};
const mapDispatchToProps = (dispatch) => {
    return {
        addTodoInfo: (id, todoInfo) => dispatch(actions.addTodoInfo(id, todoInfo)),
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(Details);
