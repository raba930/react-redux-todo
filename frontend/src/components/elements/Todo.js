import React from 'react';
import TodoListItem from './TodoListItem';
import TodoText from './TodoText';
import TodoControls from './TodoControls';

const Todo = (props) => {
    return (
        <TodoListItem completed={props.todo.completed} id={props.todo._id}>
            <TodoText>{props.todo.text}</TodoText>
            <TodoControls
                toggleComplete={props.toggleComplete}
                todoInfo={props.todoInfo}
                removeTodo={props.removeTodo}
                completed={props.todo.completed}
                index={props.index}
            >
            </TodoControls>
        </TodoListItem>
    );
};

export default Todo;
