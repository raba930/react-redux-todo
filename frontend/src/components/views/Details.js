import React, {Component} from 'react';
import DetailsHeader from '../elements/DetailsHeader';
import Button from '../elements/Button';
import TodoDetails from '../elements/TodoDetails';


class Details extends Component {
    constructor(props) {
        super(props);
        this.addInfo = this.addInfo.bind(this);
        this.goToHomepage = this.goToHomepage.bind(this);
    }
    addInfo() {
        const id = Number(this.props.match.params.id);
        const text = this.infoArea.value;
        this.props.addTodoInfo(id, text);
        this.goToHomepage();
    }
    goToHomepage() {
        this.props.history.push('/');
    }
    render() {
        return (
            <div className="detailsWrap">
                <DetailsHeader text={this.props.todo.text} completed={this.props.todo.completed}></DetailsHeader>
                <TodoDetails
                    innerRef={infoArea => { this.infoArea = infoArea; }}
                    placeholder="Aditional informations about this todo"
                    defaultValue={this.props.todo.info}>
                </TodoDetails>
                <Button right onClick={this.goToHomepage}> Cancel </Button>
                <Button right comfort onClick={this.addInfo}> Save </Button>
            </div>
        );
    }
}

export default Details;
