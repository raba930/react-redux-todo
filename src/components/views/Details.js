import React, {Component} from 'react';

class Details extends Component {
    constructor(props) {
        super(props);
        this.addInfo = this.addInfo.bind(this);
        this.goToHomepage = this.goToHomepage.bind(this);
    }
    addInfo() {
        const id = Number(this.props.match.params.id);
        const text = this.refs.infoArea.value;
        this.props.addTodoInfo(id, text);
        this.goToHomepage();
    }
    goToHomepage() {
        this.props.history.push('/');
    }
    render() {
        return (
            <div className="detailsWrap">
                <h3>
                    <span className="todo-text">{this.props.todo.text}</span>
                    {this.props.todo.completed && <i className="small material-icons done-icon">done</i>}
                </h3>
                <textarea ref="infoArea" defaultValue={this.props.todo.info} className="materialize-textarea" placeholder="Aditional informations about this todo" />
                <button onClick={this.goToHomepage} className="btn light-blue darken-4 cancel-btn"> Cancel </button>
                <button onClick={this.addInfo} className="btn light-blue darken-4 save-details"> Save </button>
            </div>
        );
    }
}

export default Details;
