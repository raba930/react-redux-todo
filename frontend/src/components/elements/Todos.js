import React from 'react';
import Todo from '../elements/Todo';
import getVisibleTodos from '../../reducers/getVisibleTodos';
import styled from 'styled-components';

const TodosList = styled.ul`
    margin-top: 5vh;
`;

const Todos = ({ todos, filter, removeTodo, toggleComplete }) => {
    const visibleTodos = getVisibleTodos(todos, filter);
    const parsedTodos = visibleTodos.map((todo, ind) => {
        return <Todo todo={todo} key={ind} removeTodo={removeTodo} toggleComplete={toggleComplete} />;
    });
    return (
        <TodosList>
            {parsedTodos}
        </TodosList>
    );
};

export default Todos;
