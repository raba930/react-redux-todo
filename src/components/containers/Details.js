import { connect } from 'react-redux';
import * as todoActions from '../../actions/todo';
import Details from '../views/Details';

const mapStateToProps = (state, prop) => {
    const todo = state.todo.todos.find((td, ind) => ind === Number(prop.match.params.id));
    return {
        todo: todo
    };
};
const mapDispatchToProps = (dispatch) => {
    return {
        addTodoInfo: (id, todoInfo) => dispatch(todoActions.addTodoInfo(id, todoInfo)),
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(Details);
