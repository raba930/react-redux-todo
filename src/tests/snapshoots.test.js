import React from 'react';
import renderer from 'react-test-renderer';
import Todos from '../components/elements/todos';
import Todo from '../components/elements/todo';
import Details from '../components/views/Details';
import { MemoryRouter } from 'react-router-dom';

const propSetup = filter => {
    return {
        todos: [{
            text: 'first',
            completed: false,
            info: 'f'
        },{
            text: 'second',
            completed: true,
            info: 's'
        },{
            text: 'third',
            completed: true,
            info: 't'
        }],
        filter: filter
    };
};

describe('Todos snapshots', () => {
    test('All todos', () => {
        let props = propSetup('SHOW_ALL');
        const component = renderer.create(<MemoryRouter><Todos {...props} /></MemoryRouter>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Active todos', () => {
        let props = propSetup('SHOW_ACTIVE');
        const component = renderer.create(<MemoryRouter><Todos {...props} /></MemoryRouter>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Todo snapshots', () => {
    test('Completed todo', () => {
        let props = {todo: propSetup().todos[1]};
        const component = renderer.create(<MemoryRouter><Todo {...props} /></MemoryRouter>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Not completed todo', () => {
        let props = {todo: propSetup().todos[0]};
        const component = renderer.create(<MemoryRouter><Todo {...props} /></MemoryRouter>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Details snapshots', () => {
    test('Completed todo', () => {
        let props = {todo: propSetup().todos[1]};
        const component = renderer.create(<MemoryRouter><Details {...props} /></MemoryRouter>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Not completed todo', () => {
        let props = {todo: propSetup().todos[0]};
        const component = renderer.create(<MemoryRouter><Details {...props} /></MemoryRouter>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
