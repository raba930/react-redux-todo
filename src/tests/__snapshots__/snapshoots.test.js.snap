exports[`Details snapshots Completed todo 1`] = `
<div
  className="detailsWrap">
  <h3>
    <span
      className="todo-text">
      second
    </span>
    <section
      className="small material-icons gesLkU">
      done
    </section>
  </h3>
  <textarea
    className="materialize-textarea cYOoHJ"
    defaultValue="s"
    placeholder="Aditional informations about this todo" />
  <button
    className="btn light-blue darken-4 jeYtjl"
    onClick={[Function]}>
     Cancel 
  </button>
  <button
    className="btn light-blue darken-4 cdkWzD"
    onClick={[Function]}>
     Save 
  </button>
</div>
`;

exports[`Details snapshots Not completed todo 1`] = `
<div
  className="detailsWrap">
  <h3>
    <span
      className="todo-text">
      first
    </span>
  </h3>
  <textarea
    className="materialize-textarea cYOoHJ"
    defaultValue="f"
    placeholder="Aditional informations about this todo" />
  <button
    className="btn light-blue darken-4 jeYtjl"
    onClick={[Function]}>
     Cancel 
  </button>
  <button
    className="btn light-blue darken-4 cdkWzD"
    onClick={[Function]}>
     Save 
  </button>
</div>
`;

exports[`Todo snapshots Completed todo 1`] = `
<li
  className="card-panel light-green lighten-1"
  id={undefined}>
  <span
    className="todo-text">
    second
  </span>
  <section
    className="controls gesLkU">
    <span
      className="ehlOiC"
      onClick={undefined}>
      <i
        className="small material-icons complete">
        restore
      </i>
    </span>
    <span
      className="iAwBni"
      onClick={undefined}>
      <a
        href="/details/undefined"
        onClick={[Function]}>
        <i
          className="small material-icons info">
          info_outline
        </i>
      </a>
    </span>
    <span
      className="ehEPdz"
      onClick={undefined}>
      <i
        className="small material-icons remove">
        delete
      </i>
    </span>
  </section>
</li>
`;

exports[`Todo snapshots Not completed todo 1`] = `
<li
  className="card-panel light-blue lighten-1"
  id={undefined}>
  <span
    className="todo-text">
    first
  </span>
  <section
    className="controls gesLkU">
    <span
      className="ehlOiC"
      onClick={undefined}>
      <i
        className="small material-icons complete">
        done
      </i>
    </span>
    <span
      className="iAwBni"
      onClick={undefined}>
      <a
        href="/details/undefined"
        onClick={[Function]}>
        <i
          className="small material-icons info">
          info_outline
        </i>
      </a>
    </span>
    <span
      className="ehEPdz"
      onClick={undefined}>
      <i
        className="small material-icons remove">
        delete
      </i>
    </span>
  </section>
</li>
`;

exports[`Todos snapshots Active todos 1`] = `
<ul
  className="dNIryj">
  <li
    className="card-panel light-blue lighten-1"
    id={0}>
    <span
      className="todo-text">
      first
    </span>
    <section
      className="controls gesLkU">
      <span
        className="ehlOiC"
        onClick={undefined}>
        <i
          className="small material-icons complete">
          done
        </i>
      </span>
      <span
        className="iAwBni"
        onClick={undefined}>
        <a
          href="/details/0"
          onClick={[Function]}>
          <i
            className="small material-icons info">
            info_outline
          </i>
        </a>
      </span>
      <span
        className="ehEPdz"
        onClick={undefined}>
        <i
          className="small material-icons remove">
          delete
        </i>
      </span>
    </section>
  </li>
</ul>
`;

exports[`Todos snapshots All todos 1`] = `
<ul
  className="dNIryj">
  <li
    className="card-panel light-blue lighten-1"
    id={0}>
    <span
      className="todo-text">
      first
    </span>
    <section
      className="controls gesLkU">
      <span
        className="ehlOiC"
        onClick={undefined}>
        <i
          className="small material-icons complete">
          done
        </i>
      </span>
      <span
        className="iAwBni"
        onClick={undefined}>
        <a
          href="/details/0"
          onClick={[Function]}>
          <i
            className="small material-icons info">
            info_outline
          </i>
        </a>
      </span>
      <span
        className="ehEPdz"
        onClick={undefined}>
        <i
          className="small material-icons remove">
          delete
        </i>
      </span>
    </section>
  </li>
  <li
    className="card-panel light-green lighten-1"
    id={1}>
    <span
      className="todo-text">
      second
    </span>
    <section
      className="controls gesLkU">
      <span
        className="ehlOiC"
        onClick={undefined}>
        <i
          className="small material-icons complete">
          restore
        </i>
      </span>
      <span
        className="iAwBni"
        onClick={undefined}>
        <a
          href="/details/1"
          onClick={[Function]}>
          <i
            className="small material-icons info">
            info_outline
          </i>
        </a>
      </span>
      <span
        className="ehEPdz"
        onClick={undefined}>
        <i
          className="small material-icons remove">
          delete
        </i>
      </span>
    </section>
  </li>
  <li
    className="card-panel light-green lighten-1"
    id={2}>
    <span
      className="todo-text">
      third
    </span>
    <section
      className="controls gesLkU">
      <span
        className="ehlOiC"
        onClick={undefined}>
        <i
          className="small material-icons complete">
          restore
        </i>
      </span>
      <span
        className="iAwBni"
        onClick={undefined}>
        <a
          href="/details/2"
          onClick={[Function]}>
          <i
            className="small material-icons info">
            info_outline
          </i>
        </a>
      </span>
      <span
        className="ehEPdz"
        onClick={undefined}>
        <i
          className="small material-icons remove">
          delete
        </i>
      </span>
    </section>
  </li>
</ul>
`;
